feat(dashboard): add dynamic document titles for browser tabs

bqst opened the pull request:
Closes #13868

## Summary

**What** — What changes are introduced in this PR?

Add dynamic document titles (`<title>`) to the admin dashboard, allowing users to distinguish between multiple open tabs based on the current page.

**Why** — Why are these changes relevant or necessary?  

Currently, all admin dashboard pages show the same browser tab title, making it difficult to identify tabs when multiple are open. This improves UX by showing contextual titles like "Medusa Sweatpants - Medusa" for a product detail page.

**How** — How have these changes been implemented?

- Created `useDocumentTitle` hook that extracts titles from route breadcrumbs or loader data
- Added `DocumentTitle` component using `react-helmet-async` (already configured)
- Integrated into `Shell` (for authenticated pages) and `PublicLayout` (for login, reset-password, invite)
- For detail pages, title is extracted from `match.data` (loader data) to show entity names

**Testing** — How have these changes been tested, or how can the reviewer test the feature?

- Navigate to different pages and verify the browser tab title updates correctly
- Check list pages show domain name (e.g., "Products - Medusa")
- Check detail pages show entity name (e.g., product title, order display_id)
- Check public pages (login, reset-password, invite) show correct titles
- Verify multiple tabs are distinguishable by their titles

---

## Examples

```tsx
// List page: /products
// Browser tab: "Products - Medusa"

// Detail page: /products/prod_123
// Browser tab: "Medusa Sweatpants - Medusa"

// Order detail: /orders/order_456
// Browser tab: "#12345 - Medusa"

// Login page: /login
// Browser tab: "Log in - Medusa"
```

The hook automatically extracts titles from:
- Breadcrumb functions that return strings
- Loader data for detail pages (product.title, order.display_id, customer name, etc.)

---

## Checklist

Please ensure the following before requesting a review:

- [ ] I have added a **changeset** for this PR
    - Every non-breaking change should be marked as a **patch**
    - To add a changeset, run `yarn changeset` and follow the prompts
- [ ] The changes are covered by relevant **tests**
- [x] I have verified the code works as intended locally
- [ ] I have linked the related issue(s) if applicable

---

## Additional Context

Supported entity types for detail pages:
- Product, Order, Customer, Collection, Category, Promotion, Campaign
- Price List, Customer Group, Region, Sales Channel, Stock Location
- Inventory Item, Reservation, API Key, Tax Region, User
- Product Type, Product Tag, Workflow Execution, Variant, Shipping Profile

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds contextual `<title>` updates throughout the admin.
> 
> - New `useDocumentTitle` hook derives titles from `handle.breadcrumb` or loader data (supports many entity types) and formats as `"<Page> - Medusa"`
> - New `DocumentTitle` component (Helmet) and integrated into `Shell` and `PublicLayout` so both authenticated and public pages update titles
> - Route map updates: add `handle.breadcrumb` for `/login`, `/reset-password`, and `/invite`
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit c57f307d7ba5f8e5674a78e08433dae591f995df. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

bqst committed:
feat: implement DocumentTitle component for dynamic page titles

* Add DocumentTitle component to manage document titles based on route data and breadcrumbs.
* Integrate DocumentTitle into PublicLayout and Shell components for consistent title updates.
* Enhance route configuration to include breadcrumb titles for login, reset-password, and invite routes.

bqst committed:
feat: add dynamic document titles to admin dashboard

NicolasGorga committed:
Merge branch 'develop' into feat/admin-dashboard-page-titles

NicolasGorga reviewed (commented):
Hey @bqst thanks for the contribution. If you could have a look at the PR reviewer comments and attend to them that'd be great!

bqst committed:
test(dashboard): add unit tests for useDocumentTitle hook

Address PR review feedback:
- export getTitleFromRouteData and getTitleFromBreadcrumb so the title
  extraction logic can be unit-tested as pure functions.
- 42 tests for getTitleFromRouteData covering every supported entity
  type (product, order, customer, collection, category, promotion,
  campaign, price_list, customer_group, region, sales_channel,
  stock_location, inventory_item, reservation, api_key, tax_region,
  user, product_type, product_tag, workflow_execution, variant,
  shipping_profile), fallback fields, and non-object input.
- 7 tests for getTitleFromBreadcrumb covering string returns, React
  element data fallback, missing handle, throwing breadcrumbs.
- 4 jsdom tests for the useDocumentTitle hook itself: default title,
  string breadcrumb, most-specific match wins, React element fallback.
- document the loader-data property-path coupling in the hook so future
  maintainers know to update it when a loader's shape changes.
- add the missing changeset.

bqst commented:
Addressed all three required items from @medusa-os-bot in a965e5062b:

1. **Changeset**: added `.changeset/dashboard-document-titles.md` (`@medusajs/dashboard` patch, message matches the PR title).
2. **Linked issue**: added `Closes #13868` at the top of the PR description.
3. **Tests**: 46 new vitest specs for the hook.
   - `getTitleFromRouteData` (42 tests): every supported entity type (product, order, customer, collection, category, promotion, campaign, price list, customer group, region, sales channel, stock location, inventory item, reservation, API key, tax region, user, product type, product tag, workflow execution, variant, shipping profile), fallback fields (e.g. customer full name → email, inventory item title → sku, tax region province → country), and non-object input handling.
   - `getTitleFromBreadcrumb` (7 tests): string returns, React element data fallback, missing/undefined handle, throwing breadcrumbs.
   - `useDocumentTitle` hook (4 jsdom tests): default title, string breadcrumb formatting, most-specific match wins, React element fallback.

Also added the doc comment about the loader-data property-path coupling that you flagged in the Notes section, so future maintainers know to update the hook when an entity loader's shape changes.

Full suite passes locally: 57 tests across 4 files.

bqst committed:
Merge remote-tracking branch 'upstream/develop' into feat/admin-dashboard-page-titles

bqst committed:
chore(dashboard): remove duplicate changeset

NicolasGorga reviewed (changes requested):
Thinking about this a little bit more and to avoid hardcoding the title resolution for detail routes, what do you think about the following?: 

Following current `breadcrumb` approach, extend the route handle to include an optional `seo` property

```ts
handle?: {
    breadcrumb?: (match: UIMatch) => string | ReactNode
    seo?: (match: UIMatch) => { title?: string } // new
  }
```

Just as with static routes breadcrumbs we could do the same for the seo.title

```ts
handle: {
    breadcrumb: () => t("orders.domain"),
    seo: () => ({ title: t("orders.domain") }),
  }
```

And for dynamic detail routes

```ts
handle: {
    breadcrumb: (match) => <Breadcrumb {...match} />,
    seo: (match: UIMatch<HttpTypes.AdminOrderResponse>) => ({
      title: match.data?.order ? `#${match.data.order.display_id}` : undefined,
    }),
  }
```

Then you simplify `useDocumentTitle` to

```ts
export function useDocumentTitle() {
  const matches = useMatches() as UIMatch<unknown, RouteHandle>[]

  const title = useMemo(() => {
    const titles = matches
      .map((match) => m.handle?.seo?.title)
      .filter((t): t is string => t !== null).at(-1)

    if (!title) {
      return DEFAULT_TITLE
    }

    return `${title} - Medusa`
  }, [matches])

  return title
}
```

NicolasGorga commented on packages/admin/dashboard/src/hooks/use-document-title.tsx:
```diff
@@ -0,0 +1,99 @@
+    const pageTitle = matches
+      .map((match) => getTitleFromMatch(match))
+      .filter((value): value is string => value !== null)
+      .at(-1)
```
nit: could avoid traversing the whole tree and calling the function for leaves that define seo or handle, by returning early if you do:

```ts
let pageTitle = ""

for (let indx = matches.length - 1; indx >= 0; indx--) {
   const resolvedTitle = getTitleFromMatch(matches[indx])

   if (resolvedTitle) {
      pageTitle = resolvedTitle
      break
   }
}

```

NicolasGorga commented on packages/admin/dashboard/src/dashboard-app/routes/get-route.map.tsx:
```diff
@@ -6,7 +6,7 @@ import { MainLayout } from "../../components/layout/main-layout"
 import { SettingsLayout } from "../../components/layout/settings-layout"
 import { ErrorBoundary } from "../../components/utilities/error-boundary"
-import { TaxRegionDetailBreadcrumb } from "../../routes/tax-regions/tax-region-detail/breadcrumb"
+import { TaxRegionDetailBreadcrumb, seo as taxRegionSeo } from "../../routes/tax-regions/tax-region-detail/breadcrumb"
```
What's the reason behind importing tax region seo specifically like this? Could you change it so that it follows the same pattern as other domains in this file?
